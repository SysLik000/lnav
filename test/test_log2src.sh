#! /bin/bash

export TZ=UTC
export YES_COLOR=1
unset XDG_CONFIG_HOME

run_cap_test ${lnav_test} -n \
    -c ":add-source-path ${test_dir}/log2src/python" \
    -c ";SELECT log_line, log_msg_src, log_msg_format, log_msg_values FROM all_logs" \
    -c ":write-jsonlines-to -" \
    ${test_dir}/log2src/python/python-example.0

run_cap_test ${lnav_test} -nN \
    -c ":add-source-path ${test_dir}/log2src/python" \
    -c ";SELECT * FROM source_log_stmt('log_example.py')" \
    -c ":write-jsonlines-to -"
