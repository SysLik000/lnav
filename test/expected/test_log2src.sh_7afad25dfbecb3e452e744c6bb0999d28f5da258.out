{"log_line":0,"log_msg_src":{"file":"{test_dir}/log2src/python/log_example.py","begin_line":19,"end_line":19,"name":"main","language":"Python"},"log_msg_format":"(?s)^debug message: initializing demo \\w$","log_msg_values":[]}
{"log_line":1,"log_msg_src":{"file":"{test_dir}/log2src/python/log_example.py","begin_line":20,"end_line":21,"name":"main","language":"Python"},"log_msg_format":"(?s)^info message:\\nprocessing started -- (.+)$","log_msg_values":[{"expr":"sys.argv[0]","value":"log_example.py"}]}
{"log_line":3,"log_msg_src":{"file":"{test_dir}/log2src/python/log_example.py","begin_line":22,"end_line":22,"name":"main","language":"Python"},"log_msg_format":"(?s)^warning message:\\nlow disk space$","log_msg_values":[]}
{"log_line":5,"log_msg_src":{"file":"{test_dir}/log2src/python/log_example.py","begin_line":23,"end_line":23,"name":"main","language":"Python"},"log_msg_format":"(?s)^error message: failed to open resource$","log_msg_values":[]}
{"log_line":6,"log_msg_src":{"file":"{test_dir}/log2src/python/log_example.py","begin_line":27,"end_line":27,"name":"main","language":"Python"},"log_msg_format":"(?s)^exception message: caught unexpected error$","log_msg_values":[]}
